sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/ProjectScanContainer.java

```diff
@@ -238,6 +238,10 @@ protected void doAfterStart() {
     if (StringUtils.isNotEmpty(organization)) {
       LOG.info("Organization key: {}", organization);
     }
+    String branch = tree.root().definition().getBranch();
+    if (branch != null) {
```

Comment:
getBranch() handles that case and returns null. Put I will add a medium test for it.